python/pyfeelpp-mor/feelpp/mor/reducedbasis/reducedbasis_timeOffline.py

```diff
@@ -0,0 +1,947 @@
+
+        s = ["","s"][len(res)>1]
+        print(f"[reducedbasis] POD-greedy, POD step : {len(res)} mode{s} computed")
+
+        return res
+
+    def greedyStep(self, xi_train, betas, g):
+        """Run the greedy step to select the parameter
+
+        Args:
+            xi_train (list): list of parameters used for training
+            betas (dict): dict containing the beta matrices for each parameter
+            g (function): right-hand side time-dependent function
+
+        Returns:
+            tuple: Delta_max maximal error bound, i_max index of the parameter in xi_train, mu_max parameter selected
+        """
+        mu_max = None
+        i_max = 0
+        Delta_max = -np.float('inf')
+        self.Z_to_matrix()
+        for i,mu in enumerate(tqdm(xi_train,desc=f"[reducedbasis] POD-greedy, greedy step", ascii=False, ncols=120)):
+
+            beta = betas[mu]
+
+            # /!\ Z_to_matirx must be called before calling the next line
+            # Delta = self.computeOnlineError(mu, beta, g, computeEnergyNorm=True)
+            Delta = self.computeDirectError(mu, beta, g, computeEnergyNorm=True)
+            Delta_tmp = Delta / self.EnNorm[-1]
+
+            if Delta_tmp > Delta_max:
+                i_max = i
+                mu_max = mu
+                Delta_max = Delta_tmp
+        return Delta_max, i_max, mu_max
+
+    def generateBasisPODGreedy(self, mu0, mu_train, g, eps_tol=1e-6, R=1, delta=None, doNotUseGreedy=False, 
+            check_error=False, fspace=None):
+        """Run POD(t)-Greedy(µ) algorithm
+
+        Args:
+            mu0 (float): initial parameter
+            mu_train (list of ParameterSpaceElement): parameter train set
+            g (function): right-hand side time-dependent function
+            eps_tol (float): stopping criterion.Default to 1e-6
+            R (int, optional): number of POD modes to compute.
+            delta (float, optional): representativiness of the Nm first POD modes
+            doNotUseGreedy (bool, optional): if True, the greedy step is not used. Defaults to False.
+            check_error (bool, optional): if True, the error is computed at each iteration. Defaults to False.
+            space (function, optional): function to compute the space error, advised if check_error=True Defaults to None.
+
+        Returns:
+            list of parameters selected
+        """
+        SN = []
+        self.Z = []
+
+        mu_star = mu0
+        Delta_max = 1 + eps_tol
+
+        betas = {}
+        for i,mu in enumerate(tqdm(mu_train, desc=f"[reducedbasis] Computing betas", ascii=False, ncols=120)):
+            betas[mu] = self.model.computeBetaQm(mu)
+
+        self.computeOfflineErrorRhs()
+
+        while Delta_max > eps_tol:
+
+            SN.append(mu_star)
+
+            # POD(t) step
+            if self.N == 0:
+                POD = self.computePODMode(mu_star, g, R=R, delta=delta, proj=False)
+                for ksi in POD:
+                    self.Z.append(ksi)
```

Comment:
there is still an issue when several POD modes are computed : the norm of the residual is not correct when the scalar product terms computed offline are used

Reply:
Weird. You remove the projection before applying the pod